### PR TITLE
Name, tag and save docker image

### PR DESCRIPTION
**Pull request by lissyx**

### CODE REVIEW

File: hooks/pre_build [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# Docker Hub expects an existing ready-to-use Dockerfile, and this hook allows
+# to generate one basing on the existing |make| template, but we need to change
+# in-place, hence why the |cp| statement
+#
+# Docker Hub Automated Builds needs to be configured as:
+#
+# Source Type | Source                | Docker Tag | Dockerfile Location   | Build Context
+# ----------------------------------------------------------------------------------------
+# Tag         | /^v([0-9.\-a-zA-Z]+)/ | v{\1}      | Dockerfile.train.tmpl | /
+# Tag         | /^v([0-9.\-a-zA-Z]+)/ | v{\1}      | Dockerfile..tmpl | /
```
Comment (reuben) [inline hooks/pre_build:15]: Missing `build` in this line on the `Dockerfile location` table?

File: hooks/pre_build [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+set -e # Exit immediately if a command exits with a non-zero status.
+set -u # Treat unset variables as an error.
+
+DOCKERFILE_TARGET=$(echo ${DOCKERFILE_PATH} | sed -e 's/\.tmpl//g')
```
Comment (reuben) [inline hooks/pre_build:20]: How is `DOCKERFILE_PATH` populated? Maybe add a brief comment?

File: hooks/pre_build (hunk 1 above)
Comment (lissyx) [reply hooks/pre_build]: Yeah, I wrote it and it disappeared :(

File: hooks/pre_build (hunk 2 above)
Comment (lissyx) [reply hooks/pre_build:25]: Done, explanation on env vars + link to the doc

**Pull request merged**